paths/paths.go

```diff
@@ -59,3 +59,34 @@ func DefaultTailscaledStateFile() string {
 	}
 	return ""
 }
+
+// MkStateDir ensures that the directory dirPath exists and has the correct
+// permisions:
+// If it creates the directory, then the directory is created with the correct permissions.
+// If the directory already exists, we ensure that the permissions are set
+// correctly.
+func MkStateDir(dirPath string) error {
+	info, err := os.Stat(dirPath)
+	if err == nil && !info.IsDir() {
+		// dirPath already exists but is not a directory
+		return os.ErrInvalid
+	} else {
```

Comment:
Go style is to do early returns and outdent the rest, without much nesting of elses.